### PR TITLE
Improve setup validation

### PR DESCRIPTION
**Pull request by hlb8122**
**Motivation**
A few people had expressed problems when traversing the setup phase. The origin of these problems were due to lax validation on input fields.

**Changes**
- Add validation on Profile and Inbox data during setup.
- Split `defaultRelayData` in `pendingRelayData` and `defaultRelayData` - the former for a dummy contact profile used while adding contacts, the latter for an empty profile.

### CODE REVIEW

File: src/pages/Setup.vue [hunk 1]
```diff
@@ -536,6 +536,12 @@ export default {
   computed: {
     electrumConnected () {
       return this.$electrum.connected
+    },
+    isWalletValid () {
+      return ((this.seedData.type === 'new') || (this.seedData.type === 'import' && this.seedData.valid))
+    },
+    isRelayValid () {
+      return [this.relayData.profile.name && this.relayData.profile.avatar && this.relayData.inbox.acceptancePrice].every(Boolean)
```
Comment (schancel) [inline src/pages/Setup.vue:544]: What's going on here?  This looks like rust not JS.

This should be enough.
```javascript
this.relayData.profile.name && this.relayData.profile.avatar && this.relayData.inbox.acceptancePrice
```

File: src/pages/Setup.vue [hunk 2]
```diff
@@ -536,6 +536,12 @@ export default {
   computed: {
     electrumConnected () {
       return this.$electrum.connected
+    },
+    isWalletValid () {
+      return ((this.seedData.type === 'new') || (this.seedData.type === 'import' && this.seedData.valid))
```
Comment (schancel) [inline src/pages/Setup.vue:541]: Why do we need this?  Can we click continue right now with garbage data?

File: src/utils/constants.js [hunk 3]
```diff
@@ -21,6 +21,18 @@ export const defaultAcceptancePrice = 100
...
 export const relayUrlOptions = ['34.68.170.199:8531']
 export const defaultRelayData = {
+  profile: {
+    name: '',
+    bio: null,
+    avatar: null,
+    pubKey: null
+  },
+  inbox: {
+    acceptancePrice: 10_000
```
Comment (schancel) [inline src/utils/constants.js:31]: ? this is above the default stamp price right now?

File: src/pages/Setup.vue (hunk 2 above)
Comment (hlb8122) [reply src/pages/Setup.vue:541]: Yeah, people were leaving the fields empty.

**Pull request merged**